lib/Subscription/Manage.php

```diff
@@ -111,12 +117,8 @@ private function updateSubscriptions(Subscriber $subscriber, array $subscriberDa
     if ($allowedSegments) {
       $segmentsIds = array_intersect($segmentsIds, $allowedSegments);
     }
-    foreach ($segmentsIds as $segmentId) {
-      SubscriberSegment::createOrUpdate([
-        'subscriber_id' => $subscriber->id,
-        'segment_id' => $segmentId,
-        'status' => Subscriber::STATUS_SUBSCRIBED,
-      ]);
+    if ($segmentsIds) {
+      $this->subscriberActions->subscribe($subscriberData, $segmentsIds);
```

Comment:
I think we can't use `$this->subscriberActions->subscribe` in this case. If we keep it as is it would trigger subscription confirmation email, new subscriber notification email and also welcome email scheduling each time the subscriber changes something in the manage subscription form.
The specification mentions that we should schedule the welcome email when subscriber subscribes on manage subscription page. I think that this means that we should do these actions only when a subscriber subscribes to a new list.